tests/test_qir/test_compile.py

```diff
@@ -0,0 +1,542 @@
+# Copyright 2023 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import inspect
+import re
+from operator import attrgetter
+from typing import Callable
+
+import pytest
+
+from dwave.gate import Circuit
+from dwave.gate import operations as ops
+
+pyqir = pytest.importorskip("pyqir")
+# QIR submodule imports must postcede PyQIR importorskip
+from dwave.gate.qir.compiler import BaseModule, CompileError, qir_module  # noqa: E402
+from dwave.gate.qir.instructions import InstrType, Instruction, Operations  # noqa: E402
+
+
+@pytest.fixture(scope="function")
+def two_op_module():
+    """Module with an ``X``, a ``CNOT`` and a ``Measurement`` operation."""
+    mod = BaseModule("QIR_module", 2, 1)
+
+    args_op_0 = (mod.qubits[0],)
+    args_op_1 = (mod.qubits[0], mod.qubits[1])
+    args_meas = (mod.qubits[0], mod.results[0])
+
+    instruction_0 = Instruction(InstrType.BASE, "x", args_op_0)
+    instruction_1 = Instruction(InstrType.BASE, "cx", args_op_1)
+    instruction_2 = Instruction(InstrType.MEASUREMENT, "mz", args_meas)
+
+    mod.add_instruction("body", instruction_0)
+    mod.add_instruction("body", instruction_1)
+    mod.add_instruction("measurements", instruction_2)
+
+    return mod
+
+
+def assert_equal(str_0: str, str_1: str) -> bool:
+    """Assert that two multi-line strings are equal apart from newlines
+    and spaces padding lines."""
+    str_0_split = [line for line in str_0.splitlines() if line != '']
+    str_1_split = [line for line in str_1.splitlines() if line != '']
+    assert len(str_0_split) == len(str_1_split)
+
+    for r, e in zip(str_0_split, str_1_split):
+        # strip whitespace from each line and check
+        assert r.strip() == e.strip()
```

Comment:
Still incorrect. Consider:
```
assert_equal("  ", "")
# or
assert_equal("a \n   ", "a")
```
i.e. a blank line with just spaces should be ignored.